blog/_posts/2022-07-18-secure-your-dependencies-on-github.md

```diff
@@ -0,0 +1,139 @@
+
+A similar action for the Mill build tool, the [mill-dependency-submission][mill-dependency-submisison] action, developed by Chris Kipp, is also available.
+Other solutions are available for Maven and Gradle as well.
+
+The next sections explain the motivation behind this project, show how to get started, compare Dependabot and Scala Steward, and list similar solutions for other build tools.
+
+## GitHub Dependabot and the Scala ecosystem
+
+Last year, the [Log4j CVE](https://scala-lang.org/blog-detail/2021/12/16/state-of-log4j-in-scala-ecosystem.html) reminded us of the importance of staying alert to security vulnerabilities.
+Fixing all known vulnerabilities once does not make your project secure, since critical vulnerabilities can always be found and exploited in open source libraries, even the most widely used ones.
+Being able to identify new vulnerabilities and to respond to them quickly reduces the risk of exposure to its minimum.
+But the task can be time consuming, unless we automate it.
+
+GitHub Dependabot is an automation tool that helps us secure our projects.
+Its initial approach is to scan static configuration files such as POM files or package-lock.json files to extract the dependencies of a repository and cross reference them with a database of vulnerabilities.
+But some build definitions, such as build.sbt files, are programs that cannot be statically analyzed.
+A few weeks ago, GitHub released the beta version of the Dependency submission API, a REST API that we can use in CI workflows to submit complete snapshots of dependencies when they are resolved at build-time.
+
+As a large part of the Scala community uses GitHub to host their projects, we developed sbt-dependency-submission, a GitHub action that can extract the graphs of dependencies of sbt projects, for Dependabot to analyze them and report any vulnerability.
+With this tool, we hope to contribute to building a more secure Scala open source ecosystem.
+
+## Setting up the workflow
+
+Before installing the dependency submission action, you need to ensure that the `Dependency Graph` view is enabled in the `Settings > Code security and analysis` tab of your GitHub repository.
+You should also consider enabling `Dependabot alerts`.
+
+![Dependency Graph](/resources/img/blog/github/dependency-graph.jpg)
+
+To install the action, add a new workflow in the `.github/workflows` folder, with the following definition:
+
+```yaml
+# .github/workflows/dependency-graph.yml
+name: Update Dependency Graph
+
+on:
+  push:
+    branches:
+      - main # default branch of the project
+
+# The API requires write permission on the repository to submit dependencies
```

Comment:
This is by default though isn't it? For example in the mill action I don't have any permissions set and it still works fine.